In [83]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sbs
import random
import scipy.stats as stat

In [84]:
data=pd.read_csv('C:/Users/Dell/Downloads/Sales_add.csv')

In [85]:
data

Month      Region      Manager  Sales_before_digital_add(in $)  \
0    Month-1  Region - A  Manager - A                          132921   
1    Month-2  Region - A  Manager - C                          149559   
2    Month-3  Region - B  Manager - A                          146278   
3    Month-4  Region - B  Manager - B                          152167   
4    Month-5  Region - C  Manager - B                          159525   
5    Month-6  Region - A  Manager - B                          137163   
6    Month-7  Region - C  Manager - C                          130625   
7    Month-8  Region - A  Manager - A                          131140   
8    Month-9  Region - B  Manager - C                          171259   
9   Month-10  Region - C  Manager - B                          141956   
10  Month-11  Region - B  Manager - C                          159339   
11  Month-12  Region - A  Manager - A                          178939   
12  Month-13  Region - B  Manager - A                          145062   
13  Month-14  Region - A  Manager - A                          151514   
14  Month-15  Region - C  Manager - A                          147463   
15  Month-16  Region - A  Manager - B                          177195   
16  Month-17  Region - A  Manager - C                          140862   
17  Month-18  Region - C  Manager - B                          167996   
18  Month-19  Region - B  Manager - A                          132135   
19  Month-20  Region - A  Manager - B                          152493   
20  Month-21  Region - B  Manager - A                          147425   
21  Month-22  Region - A  Manager - C                          130263   

    Sales_After_digital_add(in $)  
0                          270390  
1                          223334  
2                          244243  
3                          231808  
4                          258402  
5                          256948  
6                          222106  
7                          230637  
8                          226261  
9                          193735  
10                         203175  
11                         276279  
12                         205180  
13                         253131  
14                         229336  
15                         187305  
16                         234908  
17                         191517  
18                         227040  
19                         212579  
20                         263388  
21                         243020

In [86]:
data.describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

# Whether there is any increase in sales afterstepping into digital marketing

Null Hypothesis, H0 : µ1 >= µ2 ;no increase in sales afterstepping into digital marketing

Alternative Hypothesis, H1 : µ1 < µ2 ;increase in sales afterstepping into digital marketing


The Confidence level for this test will be 95% & we'll set  Significance level as alpha = 0.05.

In [90]:
before_sales=data['Sales_before_digital_add(in $)'].describe()
before_sales

count        22.000000
mean     149239.954545
std       14844.042921
min      130263.000000
25%      138087.750000
50%      147444.000000
75%      157627.500000
max      178939.000000
Name: Sales_before_digital_add(in $), dtype: float64

In [92]:
after_sales=data['Sales_After_digital_add(in $)'].describe()
after_sales

count        22.000000
mean     231123.727273
std       25556.777061
min      187305.000000
25%      214960.750000
50%      229986.500000
75%      250909.000000
max      276279.000000
Name: Sales_After_digital_add(in $), dtype: float64

In [99]:
data.shape

(22, 5)

size of the sample is lessthan 30 ,we perform t-test

In [101]:
t_score=stat.ttest_rel(after_sales ,before_sales,alternative="greater")
t_score

Ttest_relResult(statistic=4.730357958505286, pvalue=0.0010658314673505525)

Since the p value is 0.001 which is less than 0.05(Significance Level), We reject H0(Null Hypothesis).

Therefore, We identified that increase in sales afterstepping into digital marketing.

# The company needs to check whether there is any dependency between thefeatures “Region” and “Manager”.

Null Hypothesis, H0 : µ1 >= µ2 ;There is no dependency between the features Region and Manager.



Alternative Hypothesis, H1 : µ1 < µ2 ;There is dependency between the features Region and Manager.




The Confidence level for this test will be 95% & we'll set the level of Significance as alpha = 0.05

In [103]:
data['Region'].value_counts()

Region - A    10
Region - B     7
Region - C     5
Name: Region, dtype: int64

In [104]:
data['Manager'].value_counts()

Manager - A    9
Manager - B    7
Manager - C    6
Name: Manager, dtype: int64

Region and Manager have Categorical Data. So we perform chi-square test here.

In [107]:
ct= pd.crosstab(data.Region, data.Manager)
ct

Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A            4            3            3
Region - B            4            1            2
Region - C            1            3            1

In [108]:
chivalue,p,dof,expectedvalues=stat.chi2_contingency(ct)
print("Chi-Square Test Value=",chivalue)
print("P-score=",p)
print("Degree of Freedom=",dof)

Chi-Square Test Value= 3.050566893424036
P-score= 0.5493991051158094
Degree of Freedom= 4


Since the p score is 0.549 which is greater than 0.05. We reject H1(Alternative Hypothesis). ie; we accept H0(Null hypothesis)

Therefore , we identified that the Manager and Region features are independent. There is no dependency between the features.